src/sentry/models/files/file.py

```diff
@@ -29,8 +29,9 @@ class Meta:
         db_table = "sentry_file"
 
     def _blob_index_records(self) -> Sequence[FileBlobIndex]:
-        return list(
-            FileBlobIndex.objects.filter(file=self).select_related("blob").order_by("offset")
+        return sorted(
+            FileBlobIndex.objects.filter(file=self).select_related("blob").order_by("offset"),
```

Comment:
May as well remove sorting by offset here if we're just going to resort in memory